### ISSUE TITLE
Q: Which lib type is suitable for app configuration?

### ISSUE THREAD

**Issue by EvAlex** [manfredsteyer/ddd-bk#9]
I greatly appreciate your book and ng conf talks. I believe the topics you highlight are very important in enterprise development. 

Now I attempt to apply your ideas and am facing some problems. I would be very grateful if you provide some piece of advice. Thank you in advance.

### The problem

Say we implement a feature `signup` in `auth` domain. It requires some data access. So we make some libs: `auth-feature-signup`, `auth-data-access`, `auth-shell`. Now we start implementing data access and soon we need a way to access configuration. It may be `environment.ts` file or any other mechanism. We cannot import `environment.ts` directly as it belongs to app. We need to extract configuration to a common dependency. But which type of dependency is it?

### Considered options

**1. `util`.** 
pros:
- configuration will be accessible from anywhere
cons:
- the "utility" word usually means some common sense library function. But the configuration is very specific to application infrastructure. Moving configuration to util may introduce invalid global dependencies in enterprise monorepo where different apps are developed

**2. `feature` in config domain**
cons:
- if feature libs represent applcation layer, then this is not correct. Configuration should be in insfrastructure layer

**3. `data-access` in config domain**
pros:
- seems like correct place as configuration do belong to infrastructure layer
cons:
- the library is called "data-access", but actually it does not have any data access in it. It has configuration for other data access libraries. It is frustrating

**4. Library without a type in shared domain**
pros:
- accessible from everywhere
- no conceptual conflicts with library type
cons:
- violates the system by introducing new library type without name

**Comment by waysharknite** [comment]
This might be a solution
https://github.com/nrwl/nx/issues/208#issuecomment-475911897

**Comment by waysharknite** [comment]
You simply don't need a separate package for the application configuration, it's not suitable for that purpose.
App-specific configuration can be provided using DI as described in previous comment.
Library-specific configuration should be defined in the library package itself, then you can use it to provide that configuration using DI.

**Comment by EvAlex** [comment]
@in5anity it doesn't change anything. I still have to reference that Environment class. You may propose introducing an injection token and an interface, but still I would have to to reference them. And my initial question is the same: where is the best place for them?

**Comment by waysharknite** [comment]
This kind of configuration (http, UI) is not a part of the domain model. You can create external library outside domain layer and put it there.

**Comment by EvAlex** [comment]
@in5anity yes and this is my considered option number 4. I guess I will stick to it until we find more proper way
